### PR TITLE
Replace 'get_authenticated_user' with isMe filter

### PR DESCRIPTION
**Pull request by RashaMou**
This eliminates unnecessary api calls and makes everything a little speedier

### CODE REVIEW

**Comment by RashaMou** [comment]
@rjbs this is tested and ready for review!

File: lib/Synergy/Reactor/Linear.pm [hunk 1]
```diff
@@ -290,21 +287,19 @@ command urgent => {
...
-          header => "Urgent issues for you",
-          linear => $linear,
-        }
-      );
-    });
+    $self->_handle_search(
+      $event,
+      {
+        search => {
+          assignee => { isMe => {eq => \1} },
```
Comment (rjbs) [inline lib/Synergy/Reactor/Linear.pm:294]:
```suggestion
          assignee => { isMe => { eq => \1 } },
```

Style nit: spaces inside braces.

File: lib/Synergy/Reactor/Linear.pm [hunk 2]
```diff
@@ -421,9 +416,7 @@ command agenda => {
                     }
                   }
                 })
-              : $linear->get_authenticated_user->then(sub ($user) {
-                  return Future->done(assignee => $user->{id});
-                });
+              : Future->done(assignee => { isMe => {eq => \1} });
```
Comment (rjbs) [inline lib/Synergy/Reactor/Linear.pm:419]:
```suggestion
              : Future->done(assignee => { isMe => { eq => \1 } }); 
```

**Review by rjbs — APPROVED** [review]
Looks good!  Two nits to style, with suggestions.

**Pull request merged**